src/MatrixClient.cc

```diff
@@ -1021,20 +1027,19 @@ MatrixClient::makeUploadRequest(const QString &filename)
         endpoint.setPath(mediaApiUrl_ + "/upload");
         endpoint.setQuery(query);
 
-        QFile file(filename);
-        if (!file.open(QIODevice::ReadWrite)) {
-                qDebug() << "Error while reading" << filename;
+        if (!iodev->open(QIODevice::ReadWrite)) {
+                qDebug() << "Error while reading buffer" << iodev.data();
```

Comment:
Logged errors should contain [the specific error message](https://doc.qt.io/qt-5/qiodevice.html#errorString).